### ISSUE TITLE
train failed when EMA mode is off

### ISSUE THREAD

**Issue by frozenzo** [kekmodel/FixMatch-pytorch#42]
I found that this is because the _model.train()_  did not open again when evaluation ends.
solution: just move _mode.train()_ to the epoch loop:

- model.train()
for epoch in range(args.start_epoch, args.epochs):

->     

for epoch in range(args.start_epoch, args.epochs):
+        model.train()

**Comment by frozenzo** [comment]
dear author, could you kindly show results that **did not** use EMA for testing?

**Comment by DingYuan0118** [comment]
save question here. Considering the long training time, I wanna know did you fix the training for the fixed version?

**Comment by kekmodel** [comment]
It's late, but I'll check.

**Comment by TonyLianLong** [comment]
I met exactly the same thing: on the second epoch, the loss becomes `nan` without modification when EMA is off.

**Comment by dahuaxiya** [comment]
> 

yes, I met the same thing too. If test the model, the loss will be nan when ema is closed.However, the loss is normal without test step.

**Comment by yiwengxie** [comment]
I face the same problem mentioned above, but solved just to put the `model.train()` in the training loop:
From:
``` python
model.train()
for epoch in range(args.start_epoch, args.epochs):
```
To:
```python
for epoch in range(args.start_epoch, args.epochs):
    model.train()
```
